### DISCUSSION TITLE
Support loading of client-side index configuration for Firestore

### DISCUSSION THREAD

**Discussion by puf** [firebase/flutterfire#9913] [Feature request]
Cloud Firestore SDKs for Apple platforms (version [9.6.0](https://firebase.google.com/support/release-notes/ios#version_960_-_september_13_2022)), Android (version [24.1.0](https://firebase.google.com/support/release-notes/android#firestore_v24-1-0)), and web (version [9.9.2](https://firebase.google.com/support/release-notes/js#version_992_-_august_04_2022)) now support defining and using indexes on their underlying client-side cache which significantly improves offline query performance. The Flutter SDK should wrap these calls and expose them as a single API to Flutter developers.

**Comment by russellwheatley** [comment]
Hey @puf, this feature will be released tomorrow. PR: https://github.com/firebase/flutterfire/pull/9928

**Reply by deepak786** [reply]
So now we can speed up the offline queries. That's great.
But need to replicate the index configuration at the client side from what is configured in the Firebase console, right?

**Reply by puf** [reply]
Correct. You'll indeed need to define the indexes for the client, same as you need to define them for the server. You can find the full documentation for defining client-side indexes here: https://firebase.google.com/docs/firestore/manage-data/enable-offline#configure_offline_query_indexes

**Comment by puf** [comment]
Awesome! Thanks for the quick turnaround Russell! 🙏
Let me update my video script to also mention Flutter in that case. :)

**Comment by Mijawel** [comment]
Follow up question. When using setIndexConfiguration, does this need to be run every time on startup? And does it recreate the indexes every time it is run?

I'm asking because the documentation linked is a bit different from how offline indexes are created on flutter.
On flutter, you define the index config:
setIndexConfiguration()

On Android and iOS, you:
enableIndexAutoCreation()

Which implies that the two functions work a bit differently (flutter sets it's own local config, and android and ios pull the config from the server).
